packages/ui/src/components/repo-summary.tsx

```diff
@@ -0,0 +1,154 @@
+import { useNavigate } from 'react-router-dom'
+
+import {
+  Avatar,
+  AvatarFallback,
+  AvatarImage,
+  ButtonGroup,
+  Icon,
+  Spacer,
+  StackedList,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  Text
+} from '@/components'
+import { getInitials } from '@/utils/utils'
+
+import { CommitCopyActions } from './commit-copy-actions'
+
+export enum SummaryItemType {
+  Folder = 0,
+  File = 1
+}
+
+interface UserProps {
+  name: string
+  avatarUrl?: string
+}
+
+export interface FileProps {
+  id: string
+  type: SummaryItemType
+  name: string
+  lastCommitMessage: string
+  timestamp: string
+  user?: UserProps
+  sha?: string
+  path: string
+}
+
+interface PageProps {
+  latestFile: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'>
+  files?: FileProps[]
+}
+
+export const TopTitle = ({ file }: { file: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'> }) => {
+  const { user, lastCommitMessage } = file
+
+  return (
+    <ButtonGroup.Root verticalAlign="center" spacing="2">
+      <Avatar size="6">
+        <AvatarImage src={user?.avatarUrl} />
+        <AvatarFallback>
+          <Text size={0} color="tertiaryBackground">
+            {getInitials(user?.name || '')}
+          </Text>
+        </AvatarFallback>
+      </Avatar>
+      <Text size={2} weight="normal" color="tertiaryBackground" wrap="nowrap">
+        {user?.name}
+      </Text>
+      <Text size={2} weight="normal" color="primary" className="line-clamp-1">
+        {lastCommitMessage}
+      </Text>
+    </ButtonGroup.Root>
+  )
+}
+
+export const TopDetails = ({ file }: { file: Pick<FileProps, 'user' | 'lastCommitMessage' | 'timestamp' | 'sha'> }) => {
+  const { sha, timestamp } = file
+  return (
+    <ButtonGroup.Root verticalAlign="center" spacing="2">
+      <CommitCopyActions sha={sha || ''} />
+      <Text size={2} weight="normal" color="tertiaryBackground">
+        |
+      </Text>
+      <Text size={2} weight="normal" color="tertiaryBackground">
+        {timestamp}
+      </Text>
+    </ButtonGroup.Root>
+  )
+}
+
+export const Summary = ({ ...props }: PageProps) => {
+  const { latestFile, files } = props
```

Comment:
No need to use destructuring and rest to just put all the contents of props into props. Either just use it directly, or destructure in the function arguments.
```suggestion
export const Summary = (props: PageProps) => {
  const { latestFile, files } = props
```

or

```suggestion
export const Summary = ({ latestFile, files }: PageProps) => {
```

Reply:
Also, use `FC` to make sure the return type is set for you.
```suggestion
export const RepoSummary: FC<RepoSummaryProps> = ({ latestFile, files }) => {
```